### PR TITLE
feat: casbin upgraded to v2

### LINKED ISSUE #605: Casbin dependency update
# Summary

I found you guys still using casbin v1, but it no longer supported any more, so I ask updating casbin to v2.

# Motivation

https://github.com/devtron-labs/devtron/issues/596
https://github.com/devtron-labs/devtron/pull/598
https://github.com/casbin/casbin/issues/845
We could not optimise for you.

# Proposal

I'm here to ask do you want to update casbin to newest version?

AB#393

### LINKED ISSUE #5353: Enhancement: Upgrade Casbin version to v2
### 🔖 Feature description

Upgrade Casbin version to v2

### 🎤 Pitch / Usecases

Upgrade Casbin version to v2

### 🔄️ Alternative

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

AB#10027

### PR DESCRIPTION
**Pull request by Shivam-nagar23**
# Description

Enhancements:- Added support for casbin v2 version based on env variable
Fixes #5353 
Fixes #605  

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/auth/authorisation/casbin/Adapter.go [hunk 1]
```diff
@@ -81,6 +110,47 @@ func Create() (*casbin.SyncedEnforcer, error) {
...
+	//Adapter
+
+	auth, err1 := casbinv2.NewSyncedEnforcer("./auth_model.conf", a)
+	if err1 != nil {
+		log.Println(err)
+		return nil, err
+	}
+	e2 = auth
+	err = e2.LoadPolicy()
+	log.Println("v2 casbin Policies Loaded Successfully")
```
Comment (nishant-d) [inline pkg/auth/authorisation/casbin/Adapter.go:144]: log after error handling

File: pkg/auth/authorisation/casbin/Adapter.go [hunk 2]
```diff
@@ -119,24 +205,43 @@ func AddPolicy(policies []Policy) []Policy {
...
 func LoadPolicy() {
 	defer handlePanic()
-	err := enforcerImplRef.ReloadPolicy()
+	isCasbinV2, err := enforcerImplRef.ReloadPolicy()
 	if err != nil {
 		fmt.Println("error in reloading policies", err)
 	} else {
-		fmt.Println("policy reloaded successfully")
+		if isCasbinV2 {
+			fmt.Println("V2 policy reloaded successfully")
```
Comment (nishant-d) [inline pkg/auth/authorisation/casbin/Adapter.go:213]: let v1 and v2 ReloadPolicy() method log their identity instead logging here

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: pkg/auth/authorisation/casbin/Adapter.go (hunk 1 above)
Comment (Shivam-nagar23) [reply pkg/auth/authorisation/casbin/Adapter.go:144]: Done.

File: pkg/auth/authorisation/casbin/Adapter.go (hunk 2 above)
Comment (Shivam-nagar23) [reply pkg/auth/authorisation/casbin/Adapter.go:213]: Done.

**Review by vikramdevtron — APPROVED** [review]

**Review by nishant-d — APPROVED** [review]

**Pull request merged**